desc/plotting.py

```diff
@@ -3764,6 +3785,141 @@ def plot_field_lines_real_space(
+    ncontours : int, optional
+        Number of contours to plot.
+    fieldlines : int, optional
+        Number of (linearly spaced) magnetic fieldlines to plot. Default is 0 (none).
+    ax : matplotlib AxesSubplot, optional
+        Axis to plot on.
+    return_data : bool
+        if True, return the data plotted as well as fig,ax
+    **kwargs : dict, optional
+        Specify properties of the figure, axis, and plot appearance e.g.::
+
+            plot_X(figsize=(4,6),cmap="plasma")
+
+        Valid keyword arguments are:
+
+        * ``figsize``: tuple of length 2, the size of the figure (to be passed to
+          matplotlib)
+        * ``cmap``: str, matplotlib colormap scheme to use, passed to ax.contourf
+        * ``levels``: int or array-like, passed to contourf
+        * ``title_fontsize``: integer, font size of the title
+        * ``xlabel_fontsize``: float, fontsize of the xlabel
+        * ``ylabel_fontsize``: float, fontsize of the ylabel
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        figure being plotted to
+    ax : matplotlib.axes.Axes or ndarray of Axes
+        axes being plotted to
+    plot_data : dict
+        dictionary of the data plotted, only returned if ``return_data=True``
+
+    Examples
+    --------
+    .. image:: ../../_static/images/plotting/plot_omnigenous_field.png
+
+    .. code-block:: python
+
+        from desc.plotting import plot_omnigenous_field
+        fig, ax = plot_omnigenous_field(field)
+
+    """
+    if grid is None:
+        grid = LinearGrid(rho=[1.0], theta=101, zeta=101, endpoint=False, NFP=field.NFP)
+
+    title_fontsize = kwargs.pop("title_fontsize", None)
+    xlabel_fontsize = kwargs.pop("xlabel_fontsize", None)
+    ylabel_fontsize = kwargs.pop("ylabel_fontsize", None)
+
+    # compute data from omnigenous magnetic field
+    data = field.compute(
+        ["theta_B", "zeta_B", "|B|"], grid=grid, helicity=field.helicity, iota=iota
+    )
+    B = data["|B|"]
+    theta_B = np.mod(data["theta_B"], 2 * np.pi)
+    zeta_B = np.mod(data["zeta_B"], 2 * np.pi / field.NFP)
+
+    fig, ax = _format_ax(ax, figsize=kwargs.pop("figsize", None))
+    divider = make_axes_locatable(ax)
+
+    contourf_kwargs = {
+        "norm": matplotlib.colors.Normalize(),
+        "levels": kwargs.pop(
+            "levels", np.linspace(np.nanmin(B), np.nanmax(B), ncontours)
+        ),
+        "cmap": kwargs.pop("cmap", "jet"),
+        "extend": "both",
+    }
+
+    assert (
+        len(kwargs) == 0
+    ), f"plot_boozer_surface got unexpected keyword argument: {kwargs.keys()}"
+
+    cax_kwargs = {"size": "5%", "pad": 0.05}
+
+    if fill:
+        im = ax.tricontourf(zeta_B, theta_B, B, **contourf_kwargs)
+    else:
+        im = ax.tricontour(zeta_B, theta_B, B, **contourf_kwargs)
+    cax = divider.append_axes("right", **cax_kwargs)
+    cbar = fig.colorbar(im, cax=cax)
+    cbar.update_ticks()
+
+    if fieldlines:
```

Comment:
this is copy/pasted from the plot_boozer_surface, is there any way to combine these two since they are doing the same thing? like even if you keep this to just have this sub function do everything up to tricontour, then return the fig,ax and then do the fieldlines logic in the main plot_boozer_surface function?